src/core/model_event_queue.jl

```diff
@@ -144,17 +144,25 @@ can be evolved in time.
     Pull Requests that implement this are welcomed!
 """
 function EventQueueABM(
-        ::Type{A}, events,
+        A::Type, events,
         space::S = nothing;
         container::Type = Dict,
         properties::P = nothing,
         rng::R = Random.default_rng(),
         warn = true,
         autogenerate_on_add = true,
         autogenerate_after_action = true,
-    ) where {A<:AbstractAgent,S<:SpaceType,P,R<:AbstractRNG}
-    C = construct_agent_container(container, A)
-    agents = C()
+    ) where {S<:SpaceType,P,R<:AbstractRNG}
+    if container == StructVector
+        if !(A <: SoAType)
+            @warn "The agent type passed to the model constructor is of type $A but a model with a 
+            StructVector container will have agents of type SoAType{$A}. Pass this to the constructor to 
+            remove this warning." maxlog=1
+        else
+            A = A.body.parameters[1]
+        end
+    end
+    agents = construct_agent_container(container, A)
```

Comment:
is this one of the Interface function for the container API? If so we should put it in a subsection in the developer's docs.

Reply:
Oh, wow. This PR has introduced NO documentation changes. Yet it was merged and ready to be tagged as a new minor version...?